pkg/controller/subscription/subscription_controller.go

```diff
@@ -71,13 +94,14 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 var _ reconcile.Reconciler = &ReconcileSubscription{}
 
 type ReconcileSubscription struct {
-	client            k8sclient.Client
-	restConfig        *rest.Config
-	context           context.Context
-	cancel            context.CancelFunc
-	scheme            *runtime.Scheme
-	operatorNamespace string
-	mgr               manager.Manager
+	client              k8sclient.Client
+	context             context.Context
```

Comment:
We shouldn't store handles to contexts in structs, see the last paragraph in this overview: https://golang.org/pkg/context/

Reply:
I see, it got added in after rebasing off master. I'll see what are the impacts of removing the context from the structs and make the changes. 

thanks